extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -584,3 +542,14 @@ func ReadMachineConfiguration(pool extensionsv1alpha1.WorkerPool) *machinev1alph
 	}
 	return machineConfiguration
 }
+
+// Cleanup RollingUpdate condition
+// TODO(ary1992) Remove this in future release
+func (a *genericActuator) cleanupRollingUpdateCondition(ctx context.Context, worker *extensionsv1alpha1.Worker) error {
+	if c := gardencorev1beta1helper.GetCondition(worker.Status.Conditions, "RollingUpdate"); c != nil {
+		condition := gardencorev1beta1helper.RemoveConditions(worker.Status.Conditions, "RollingUpdate")
+		worker.Status.Conditions = condition
+		return a.client.Status().Update(ctx, worker)
```

Comment:
Can you use `tryUpdateStatus` here to prevent running into conflicts as reported in https://github.com/gardener/gardener-extension-provider-aws/issues/493? We can remove `tryUpdateStatus` in a later release.

Reply:
Can't we use `Patch()` instead of `Update()` here ?

Reply:
@timebertt can `Patch` be used for condition removal?

Reply:
I think, it's better to use Update() while operating on lists, because it uses optimistic locking by default.
https://github.com/gardener/gardener/blob/master/docs/development/kubernetes-clients.md#updates-patches-server-side-apply

Reply:
Update or patch with locking would be required here, as we can't use strategic merge patch for CRDs

Reply:
Replaced `update()` with patch with optimistic locking(`patch := client.MergeFromWithOptions(shoot.DeepCopy(), client.MergeFromWithOptimisticLock{})`) [commit](https://github.com/gardener/gardener/pull/5814/commits/853943cf41bf8d10db51af875e5e3f5a22e2c428).